src/plugins/yamlcpp/testmod_yamlcpp.c

```diff
@@ -0,0 +1,155 @@
+	Key * parentKey = keyNew (parent, KEY_VALUE, filepath, KEY_END);                                                                   \
+	KeySet * conf = ksNew (0, KS_END);                                                                                                 \
+	PLUGIN_OPEN ("yamlcpp")
+
+#define INIT_PLUGIN_GET(parent, filepath, errorMessage)                                                                                    \
+	INIT_PLUGIN (parent, filepath);                                                                                                    \
+	KeySet * keySet = ksNew (0, KS_END);                                                                                               \
+	succeed_if (plugin->kdbGet (plugin, keySet, parentKey) == ELEKTRA_PLUGIN_STATUS_SUCCESS, errorMessage)
+
+#define CLOSE_PLUGIN()                                                                                                                     \
+	keyDel (parentKey);                                                                                                                \
+	ksDel (keySet);                                                                                                                    \
+	PLUGIN_CLOSE ()
+
+// -- Functions ----------------------------------------------------------------------------------------------------------------------------
+
+static void test_contract (void)
+#ifdef __llvm__
+	__attribute__ ((annotate ("oclint:suppress[high ncss method]")))
+#endif
+{
+	printf ("• Retrieve plugin contract\n");
+
+	INIT_PLUGIN_GET ("system/elektra/modules/yamlcpp", "", "Could not retrieve plugin contract");
+	CLOSE_PLUGIN ();
+}
+
+static void test_read (char const * const filepath, KeySet * const expected)
+#ifdef __llvm__
+	__attribute__ ((annotate ("oclint:suppress")))
+#endif
+{
+	printf ("• Retrieve data from file “%s”\n", filepath);
+
+	INIT_PLUGIN_GET ("user/examples/yamlcpp", srcdir_file (filepath), "Unable to open or parse file");
+
+	compare_keyset (keySet, expected);
+
+	ksDel (expected);
+	CLOSE_PLUGIN ();
+}
+
+static void test_write_read (KeySet * const keySet)
+#ifdef __llvm__
+	__attribute__ ((annotate ("oclint:suppress")))
+#endif
+{
+	printf ("• Write data, read data, and compare results\n");
+
+	INIT_PLUGIN ("user/examples/yamlcpp", elektraFilename ());
+
+	// Write key set to file
+	succeed_if (plugin->kdbSet (plugin, keySet, parentKey) == ELEKTRA_PLUGIN_STATUS_SUCCESS, "Unable to write to file");
+
+	// Read written data
+	KeySet * keySetRead = ksNew (0, KS_END);
+	succeed_if (plugin->kdbGet (plugin, keySetRead, parentKey) == ELEKTRA_PLUGIN_STATUS_SUCCESS, "Unable to open or parse file");
+
+	// Compare data
+	compare_keyset (keySet, keySetRead);
+
+	// Clean up
+	ksDel (keySetRead);
+	CLOSE_PLUGIN ();
+}
+
+static void test_flat (void)
+{
+	test_read ("yamlcpp/Flat Block Mapping.yaml",
+#include "yamlcpp/Flat Block Mapping.h"
```

Comment:
After a second thought I prefer to not use spaces here: We already use - and _ within file names. If we also allow spaces, we have a new source for inconsistencies. See #1615

Reply:
Although I disagree with your decision – a space is the proper character to separate words – I updated the filenames in commit  e9f56e98.

Reply:
Thank you for fixing it!

I am not averse to spaces: the goal is to keep the number of different variants low (so that we do not have to waste time to reconsider this for every single file name). If you want you can suggest that we exclusively use spaces in #1615 

Because the file names are referred within code, they are also part of the coding conventions. #817 is also related